pkg/services/ngalert/state/manager.go

```diff
@@ -297,6 +307,42 @@ func (st *Manager) Put(states []*State) {
 	}
 }
 
+// flushState dumps the entire state to the database
+func (st *Manager) flushState(ctx context.Context) {
+	t := st.clock.Now()
+	st.log.Info("flushing the state")
+	st.cache.mtxStates.Lock()
+	defer st.cache.mtxStates.Unlock()
+	totalStates, errorsCnt := 0, 0
+	for _, orgStates := range st.cache.states {
+		for _, ruleStates := range orgStates {
+			for _, state := range ruleStates {
+				err := st.saveState(ctx, state)
+				totalStates++
```

Comment:
Should we increment `totalStates++` even if `err` is `non-nil`?

Reply:
I think so. The total states should reflect the total number of states. Together with the error counter, it provides the information.